schema/signature.go

```diff
@@ -25,3 +27,15 @@ type FunctionSignature struct {
 	// parameter if it is supported.
 	VarParam *function.Parameter
 }
+
+func (fs *FunctionSignature) Copy() *FunctionSignature {
+	newFS := &FunctionSignature{
+		Description: fs.Description,
+		Detail:      fs.Detail,
+		ReturnType:  fs.ReturnType, // TODO: deep copy needed?
+		VarParam:    fs.VarParam,   // TODO: deep copy needed?
+	}
+	newFS.Params = make([]function.Parameter, len(fs.Params))
+	copy(newFS.Params, fs.Params) // TODO: deep copy needed?
```

Comment:
@dbanck, do you think this needs a deep(er) copy?

Reply:
Yes, we usually need to make sure to explicitly copy slicing and pointing.